src/main/java/com/flansmod/common/guns/InventoryHelper.java

```diff
@@ -9,7 +9,7 @@
 /** Adds access to the InventoryPlayer stack combination methods for arbitrary inventories */
 public class InventoryHelper 
 {
-	public static boolean addItemStackToInventory(IInventory inventory, ItemStack stack, boolean creative) {
+	public static boolean addItemStackToInventory(IInventory inventory, ItemStack stack, boolean creative, boolean tryPutInUpperInventory) {
```

Comment:
e......e

I don't know what this is doing

Reply:
Basically it's adding a flag to make picked up items prefer the inventory over the hotbar. 
I think it's a good idea in general but probably needs some changes. I think it it would be good to have a config option for this, alongside an option to discard ammo below a certain threshold. Having an inventory full of basically empty mags is pretty annoying. 
In terms of the implementation, it would be better to pass the flag into getFirstEmptyStack and count the loop down instead of up if its true.

Reply:
Sure, that sounds like a good idea. I'd prefer it be in a different PR if possible though